src/bussinessLogic/ManageArray.java

```diff
@@ -137,6 +112,7 @@ private void setPracticeArray() {
         // set index
         int index = 0;
 
+        //TODO - add comment here
         for (int i = 0; i < charArrayFromSortedMap.length; i++) {
             for (int j = 0; j <= i; j++) {
                 practiceArray[index] = charArrayFromSortedMap[i];
```

Comment:
_index_ has the same value as _i_, from the first _for_ look. It means it can be easily replaced with it.

Reply:
👍

Reply:
Wrong. _Index_ does not change at the same pace with _i_. _Index_ has to cover the length of `practiceArray` which is different from `charArrayFromSortedMap` (which is covered by _i_)